### DISCUSSION TITLE
Upgrade error toLaravel 11

### DISCUSSION THREAD

**Discussion by bhattji** [laravel/framework#50801] [Q&A]
I am trying to upgrade my Laravel application to 11, but I am facing following error, Please help

composer update --ignore-platform-req=ext-zip
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - illuminate/database[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.3.4) does not satisfy that requirement.
    - illuminate/database[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.3.4) does not satisfy that requirement.    - Root composer.json requires spatie/laravel-permission ^5.10 -> satisfiable by spatie/laravel-permission[5.10.0, ..., 5.11.1].
    - Conclusion: don't install laravel/framework v11.0.2 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.3 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.4 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.5 (conflict analysis result)
PS C:\Users\Jaydeep Bhatt\laravel\11\bhattji_alumni_11> composer update --ignore-platform-req=ext-zip
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - illuminate/database[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.3.4) does not satisfy that requirement.
    - illuminate/database[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.3.4) does not satisfy that requirement.    - Root composer.json requires spatie/laravel-permission ^5.10 -> satisfiable by spatie/laravel-permission[5.10.0, ..., 5.11.1].
    - Conclusion: don't install laravel/framework v11.0.2 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.3 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.4 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.5 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.6 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.7 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.8 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.1.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.1 (conflict analysis result)
    - spatie/laravel-permission[5.10.0, ..., 5.11.1] require illuminate/database ^7.0|^8.0|^9.0|^10.0 -> satisfiable by illuminate/database[v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.16, v10.0.0, ..., v10.48.4].
    - Only one of these can be installed: illuminate/database[v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.16, v10.0.0, ..., v10.48.4, v11.0.0, ..., v11.1.0], laravel/framework[v11.0.0, ..., v11.1.0]. laravel/framework replaces illuminate/database and thus cannot coexist with it.
    - Root composer.json requires laravel/framework ^11.0 -> satisfiable by laravel/framework[v11.0.0, ..., v11.1.0].

**Comment by crynobone** [comment]
```
- illuminate/database[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.3.4) does not satisfy that requirement. - Root composer.json requires spatie/laravel-permission ^5.10 -> satisfiable by spatie/laravel-permission[5.10.0, ..., 5.11.1].
- Conclusion: don't install laravel/framework v11.0.2 (conflict analysis result)
- Conclusion: don't install laravel/framework v11.0.3 (conflict analysis result)
- Conclusion: don't install laravel/framework v11.0.4 (conflict analysis result)
- Conclusion: don't install laravel/framework v11.0.5 (conflict analysis result)
- Conclusion: don't install laravel/framework v11.0.6 (conflict analysis result)
- Conclusion: don't install laravel/framework v11.0.7 (conflict analysis result)
- Conclusion: don't install laravel/framework v11.0.8 (conflict analysis result)
- Conclusion: don't install laravel/framework v11.1.0 (conflict analysis result)
- Conclusion: don't install laravel/framework v11.0.1 (conflict analysis result)
- spatie/laravel-permission[5.10.0, ..., 5.11.1] require illuminate/database ^7.0|^8.0|^9.0|^10.0 -> satisfiable by illuminate/database[v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.16, v10.0.0, ..., v10.48.4].
```

The issue is clear here.

**Comment by ghost** [comment]
Use composer update -W

  -w, --with-dependencies                              Update also dependencies of packages in the argument list, except those which are root requirements.
  -W, --with-all-dependencies                          Update also dependencies of packages in the argument list, including those which are root requirements.

**Reply by bhattji** [reply]
I have tried with -W too, but the same problem

composer update -W --ignore-platform-req=ext-zip
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - illuminate/database[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.3.4) does not satisfy that requirement.
    - illuminate/database[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.3.4) does not satisfy that requirement.    - Root composer.json requires spatie/laravel-permission ^5.10 -> satisfiable by spatie/laravel-permission[5.10.0, ..., 5.11.1].
    - Conclusion: don't install laravel/framework v11.0.2 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.3 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.4 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.5 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.6 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.7 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.8 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.1.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.1.1 (conflict analysis result)
    - Conclusion: don't install laravel/framework v11.0.1 (conflict analysis result)
    - spatie/laravel-permission[5.10.0, ..., 5.11.1] require illuminate/database ^7.0|^8.0|^9.0|^10.0 -> satisfiable by illuminate/database[v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.16, v10.0.0, ..., v10.48.4].
    - Only one of these can be installed: illuminate/database[v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27, v9.0.0, ..., v9.52.16, v10.0.0, ..., v10.48.4, v11.0.0, ..., v11.1.1], laravel/framework[v11.0.0, ..., v11.1.1]. laravel/framework replaces illuminate/database and thus cannot coexist with it.
    - Root composer.json requires laravel/framework ^11.0 -> satisfiable by laravel/framework[v11.0.0, ..., v11.1.1].

**Reply by ghost** [reply]
edit your composer.json to have min values not strict values for illuminate/database
or remove it from the required because it is already required in laravel.

**Comment by joy2362** [comment]
hello @bhattji update spatie/laravel-permission version to 6 version 5.* not compatible with laravel 11

![Screenshot (1)](https://github.com/laravel/framework/assets/49846472/bfe172e5-c406-439c-bd4a-0c86146b01f2)
